### PR TITLE
Fix actor relation type search, refs #13326

### PR DESCRIPTION
**Pull request by mcantelon**
Changed how authority record search by relation type works so only the
relation type directly related to an authority record is returned.

### CODE REVIEW

File: plugins/arElasticSearchPlugin/lib/model/arElasticSearchActorPdo.class.php [hunk 1]
```diff
@@ -352,6 +354,59 @@ public function serializeAltNames()
...
+          $converseTerm = $relationTerm->getConverseActorRelationTerm();
+
+          if ($converseTerm !== null)
+          {
+            $converseTermId = $converseTerm->id;
+          }
+
+          // Cache result
+          self::$converseTermIds[$typeId] = $converseTermId;
+        };
```
Comment (sbreker) [inline plugins/arElasticSearchPlugin/lib/model/arElasticSearchActorPdo.class.php]: stray ';'?

Comment (mcantelon) [reply plugins/arElasticSearchPlugin/lib/model/arElasticSearchActorPdo.class.php]: Good catch!

**Comment by mcantelon** [comment]
Thanks @sbreker ... I've updated the PR.

File: apps/qubit/modules/term/actions/navigateRelatedComponent.class.php [hunk 2]
```diff
@@ -54,6 +54,7 @@ static function getEsDocsRelatedToTerm($relatedModelClass, $term, $options = [])
 
     // Allow for options to override default behavior
     $search = !empty($options['search']) ? $options['search'] : new arElasticSearchPluginQuery();
+    $search->query->setSize(10000);
```
Comment (jraddaoui) [inline apps/qubit/modules/term/actions/navigateRelatedComponent.class.php]: Is this really needed only to get the count?

File: plugins/arElasticSearchPlugin/lib/model/arElasticSearchActorPdo.class.php [hunk 3]
```diff
@@ -352,6 +354,59 @@ public function serializeAltNames()
     return $serialized;
   }
 
+  public static function serializeObjectDirectRelationTypes($actorId, $relationData)
```
Comment (jraddaoui) [inline plugins/arElasticSearchPlugin/lib/model/arElasticSearchActorPdo.class.php]: Why public and static?

File: plugins/arElasticSearchPlugin/lib/model/arElasticSearchActorPdo.class.php [hunk 4]
```diff
@@ -352,6 +354,59 @@ public function serializeAltNames()
...
+          // Look up converse term, if any
+          $converseTermId = $typeId;
+
+          $relationTerm = QubitTerm::getById($typeId);
+          $converseTerm = $relationTerm->getConverseActorRelationTerm();
+
+          if ($converseTerm !== null)
+          {
+            $converseTermId = $converseTerm->id;
+          }
```
Comment (jraddaoui) [inline plugins/arElasticSearchPlugin/lib/model/arElasticSearchActorPdo.class.php]: It's great that you cached them but I think we should use raw SQL in here, even if they are a few it should be a pretty easy query.

**Review by jraddaoui — CHANGES_REQUESTED** [review]

**Comment by jraddaoui** [comment]
Thanks @mcantelon!

Just a couple of questions and a suggestion, but feel free to disregard.

PS. Somehow this wasn't included in the initial review.

File: apps/qubit/modules/term/actions/navigateRelatedComponent.class.php (hunk 2 above)
Comment (mcantelon) [reply apps/qubit/modules/term/actions/navigateRelatedComponent.class.php]: Removed!

File: plugins/arElasticSearchPlugin/lib/model/arElasticSearchActorPdo.class.php (hunk 3 above)
Comment (mcantelon) [reply plugins/arElasticSearchPlugin/lib/model/arElasticSearchActorPdo.class.php]: Good point. Made it "protected".

File: plugins/arElasticSearchPlugin/lib/model/arElasticSearchActorPdo.class.php (hunk 4 above)
Comment (mcantelon) [reply plugins/arElasticSearchPlugin/lib/model/arElasticSearchActorPdo.class.php]: Good call.

**Comment by mcantelon** [comment]
Thanks @radda ! I've updated the PR.

**Review by jraddaoui — APPROVED** [review]
Thanks @mcantelon!

**Pull request merged**